clients/client-java/src/main/java/org/apache/gravitino/client/FilesetCatalog.java

```diff
@@ -238,7 +241,22 @@ public boolean dropFileset(NameIdentifier ident) {
   @Override
   public String getFileLocation(NameIdentifier ident, String subPath)
       throws NoSuchFilesetException {
-    throw new UnsupportedOperationException("Not implemented");
+    checkFilesetNameIdentifier(ident);
+    Namespace fullNamespace = getFilesetFullNamespace(ident.namespace());
+
+    CallerContext callerContext = CallerContext.CallerContextHolder.get();
+    Map<String, String> queryParams = Maps.newHashMap();
+    queryParams.put("subPath", subPath);
+    FileLocationResponse resp =
+        restClient.get(
+            formatFilesetRequestPath(fullNamespace) + "/" + ident.name() + "/" + "fileLocation",
```

Comment:
Is it better to add the th subPath into the URL, not a queryParam?

Reply:
Hi, I prefer to use the `queryParams` for the following reasons:
1. Currently other similar GET APIs pass parameters in the form of queryParams, like `RelationalTable.listPartitions()`;
2. I looked at the APIs of other commercial products, such as the REST API of Databricks' DBFS, and they also pass these parameters through queryParams: https://docs.databricks.com/api/workspace/dbfs/read;
3. In the unit test, I tried to directly splice the URL, but it seems that the test framework cannot mock the request, so I guess some frameworks may not recognize the method of directly splicing parameters to the URL.